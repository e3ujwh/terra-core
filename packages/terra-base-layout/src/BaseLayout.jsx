import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import 'terra-base/lib/baseStyles';
import styles from './BaseLayout.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * The component(s) that will be wrapped by `<BaseLayout />`.
   */
  children: PropTypes.node.isRequired,
  /**
   * If this is set to true, the base layout will not provide overflow auto.
   * The use case for this is if your child component will be provide the overflow auto sytling itself.
   */
  disableOverflow: PropTypes.bool,
};

const defaultProps = {
  disableOverflow: false,
};

const BaseLayout = ({
    children,
    disableOverflow,
    ...customProps
  }) => {
  const layoutClassNames = cx([
    'inner',
    { 'disable-overflow': disableOverflow },
    customProps.className,
  ]);

  // Investigate this vs the disable-overflow class.
  // return React.Children.only(this.props.children)
  return (
    <div {...customProps} className={layoutClassNames}>
      {children}
    </div>
  );
};

BaseLayout.propTypes = propTypes;
BaseLayout.defaultProps = defaultProps;

export default BaseLayout;
