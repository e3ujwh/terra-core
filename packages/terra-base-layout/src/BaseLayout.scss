body {
  height: 100vh;
  overflow: hidden;
  width: 100vw;
}

:local {
  /* stylelint-disable selector-class-pattern  */
  .inner {
    height: 100%;
    overflow: auto;
    position: relative;
    width: 100%;
  }

  .disable-overflow {
    overflow: hidden;
  }
  /* stylelint-enable selector-class-pattern  */
}
