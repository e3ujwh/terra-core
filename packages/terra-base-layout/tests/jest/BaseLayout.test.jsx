import React from 'react';
import BaseLayout from '../../src/BaseLayout';

// Snapshot Tests
it('should support rendering a string without translation', () => {
  const baseLayout = shallow(<BaseLayout>String</BaseLayout>);

  expect(baseLayout).toMatchSnapshot();
});

it('should support rendering a string without translation', () => {
  const baseLayout = shallow(<BaseLayout disableOverflow>String</BaseLayout>);

  expect(baseLayout).toMatchSnapshot();
});
