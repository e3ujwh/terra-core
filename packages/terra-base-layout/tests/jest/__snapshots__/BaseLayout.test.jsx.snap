exports[`test should support rendering a string without translation 1`] = `
<div
  className="inner">
  String
</div>
`;

exports[`test should support rendering a string without translation 2`] = `
<div
  className="inner disable-overflow">
  String
</div>
`;
